### ISSUE TITLE
optimize default random function

### ISSUE THREAD

**Issue by fabiolimace** [f4b6a3/uuid-creator#55]

**Comment by fabiolimace** [comment]
The `DefaultRandomFunction` was refactored to use a fixed pool of `SecureRandom`. The previous implementation used a thread local `SecureRandom`.

The thread pool SIZE is equal to the number of processors (or cores) available to the runtime. For example, if there's only 1 processor available, the pool size will be 1. If there are 16 processors available, the pool size will be 16. The maximum pool size is 32.

The current INDEX in the thread pool is calculated with this formula: `INDEX = CURRENT_THREAD_ID % POOL_SIZE`.

A BENCHMARK using a high number of threads (100) showed that the fixed pool performs almost the same as the thread local to generate random numbers using the SHA1PRNG algorithm.

```
-----------------------------------------------------------------
Benchmark                Mode  Cnt      Score      Error   Units
-----------------------------------------------------------------
Throughput.singleton    thrpt    3   3020,402 ±  449,643  ops/ms
Throughput.fixedpool    thrpt    3  14639,183 ± 1205,735  ops/ms  (NEW)
Throughput.threadlocal  thrpt    3  15163,027 ±  905,890  ops/ms  (OLD)
-----------------------------------------------------------------
```
Benchmark settings:
-   Number of threads: 100
-   SecureRandom algorithm: SHA1PRNG
-   Computer: JVM 8, Ubuntu 20.04, CPU i5-3330, 8G RAM.

**Comment by fabiolimace** [comment]
Benchmark files are available on this [GitHub Gist](https://gist.github.com/fabiolimace/71653c155ba465d78eee1c519bd02f3d)

**Comment by fabiolimace** [comment]
Released version v4.3.0.
